frontend/simple/views/components/Voting/Voting.vue

```diff
@@ -130,36 +153,130 @@ export default {
+          ),
+          against: template(
+            L('Keep {value}'), { value }
+          )
+        }
+      } else {
+        return {
+          for: template(
+            L('Change to {value}'), { value }
+          ),
+          against: template(
+            L('Keep {originalValue}'), { originalValue }
+          )
+        }
+      }
+    },
+    title () {
+      const { TypeInvitation, TypeRemoval, TypeIncome, TypeChangeThreshold, TypeApprovalThreshold, TypeRemovalThreshold } = HashableGroupProposal
+      const titleMap = {
+        [TypeInvitation]: L('Invite Member'),
+        [TypeRemoval]: L('Remove Member'),
+        [TypeIncome]: L('Change Mincome'),
+        [TypeChangeThreshold]: L('Update Rule: Change Threshold'),
+        [TypeApprovalThreshold]: L('Update Rule: Invite Threshold'),
+        [TypeRemovalThreshold]: L('Update Rule: Member Removal Threshold')
+      }
+      return titleMap[this.type]
+    },
+    text () {
+      const { initiator, value, originalValue } = this
+      const { TypeInvitation, TypeRemoval, TypeIncome, TypeChangeThreshold, TypeApprovalThreshold, TypeRemovalThreshold } = HashableGroupProposal
+      const textMap = {
+        [TypeInvitation]: template(
+          L('{initiator} proposed to <strong>invite {value}</strong> to the group'), {
+            initiator, value }
+        ),
+        [TypeRemoval]: template(
+          L('{initiator} proposed to <strong>remove {value}</strong> from the group'), {
+            initiator, value }
+        ),
+        [TypeIncome]: template(
+          L('{initiator} proposed to change the <strong>mincome from {originalValue} to {value}</strong>'), {
+            initiator, value, originalValue }
+        ),
+        [TypeChangeThreshold]: template(
+          L('{initiator} proposed to change the <strong>rule changing threshold from {originalValue} to {value}</strong>'), {
+            initiator, value, originalValue }
+        ),
+        [TypeApprovalThreshold]: template(
+          L('{initiator} proposed to change the <strong>member invite threshold from {originalValue} to {value}</strong>'), {
+            initiator, value, originalValue }
+        ),
+        [TypeRemovalThreshold]: template(
+          L('{initiator} proposed to change the <strong>member removal threshold from {originalValue} to {value}</strong>'), {
+            initiator, value, originalValue }
+        )
+      }
+      return textMap[this.type]
+    },
+    detailed () {
+      const { TypeChangeThreshold, TypeApprovalThreshold, TypeRemovalThreshold } = HashableGroupProposal
+      if ([TypeChangeThreshold, TypeApprovalThreshold, TypeRemovalThreshold].includes(this.type)) {
+        const originalCount = Math.ceil(this.votes.total * this.votes.originalValue)
+        const newCount = Math.ceil(this.votes.total * this.value)
+        const groupCount = this.votes.total
+        const actionMap = {
+          TypeChangeThreshold: L('change a rule'),
```

Comment:
Was this meant to be `[TypeChangeThreshold]`?

Reply:
yes! 😱